src/mlmd/LineageView.tsx

```diff
@@ -231,13 +282,29 @@ export class LineageView extends React.Component<LineageViewProps, LineageViewSt
       inputExecutionInputArtifactIds.push(event.getArtifactId());
     });
 
+    // This map ensures that output artifacts are fetched in an order that is sorted by the
+    // connected output execution.
+    const outputExecutionToOutputArtifactMap: Map<number, number[]> = new Map();
+
     const outputExecutionOutputArtifactIds: number[] = [];
+
     outputExecutionEvents.forEach((event) => {
       if (!isOutputEvent(event)) {
         return;
       }
 
-      outputExecutionOutputArtifactIds.push(event.getArtifactId());
+      const executionId = event.getExecutionId();
+      if (!outputExecutionToOutputArtifactMap.get(executionId)) {
+        outputExecutionToOutputArtifactMap.set(executionId, []);
+      }
+
+      const artifactId = event.getArtifactId();
+      outputExecutionOutputArtifactIds.push(artifactId);
+
+      const artifacts = outputExecutionToOutputArtifactMap.get(executionId);
+      if (artifacts) {
+        artifacts.push(artifactId);
+      }
```

Comment:
```suggestion
      artifacts && artifacts.push(artifactId);
```

Reply:
That style of code throws a lint error `unused expression, expected an assignment or function call` and is arguably less readable.

The lint failure is enforced by Pipelines which I'm developing this against and has well enforced style and lint guides. Until we implement https://github.com/kubeflow/metadata/issues/141 to enforce them here I'm sticking to those styles.

Reply:
I see, understood!